src/import.js

```diff
@@ -458,62 +436,74 @@ Sk.importModuleInternal_ = function (name, dumpJS, modname, suppliedPyBody, canS
         }
         // }
 
-        namestr = "new Sk.builtin.str('" + modname + "')";
-        finalcode += "\n" + co.funcname + "(" + namestr + ");";
+        finalcode += "\n" + co.funcname + ";";
 
-        modlocs = goog.global["eval"](finalcode);
+        modscope = goog.global["eval"](finalcode);
 
-        return (function finishLoading(modlocs) {
+        module["$d"] = {
+            "__name__": new Sk.builtin.str(modname),
+            "__doc__": Sk.builtin.none.none$,
```

Comment:
What happens when `mod.__doc__` has been defined in the module? I remember, that the parser currently just skips the comments, but they might be present in js modules.

Reply:
We haven't run the module body yet at this point! This code is providing default values that the module body can overwrite if it needs to. (Previously, we filled these in after eval-ing the code, in case the module didn't fill it in. But now we have an import system that takes account of `__package__` and `__path__`, we need those variables to be set before the module body runs so that `import` statements within the module are resolved correctly.